source/Octopus.Client/IOctopusAsyncRepository.cs

```diff
@@ -7,59 +10,37 @@ namespace Octopus.Client
-        IEnvironmentRepository Environments { get; }
-        IEventRepository Events { get; }
-        IFeaturesConfigurationRepository FeaturesConfiguration { get; }
-        IFeedRepository Feeds { get; }
-        IInterruptionRepository Interruptions { get; }
-        ILibraryVariableSetRepository LibraryVariableSets { get; }
-        ILifecyclesRepository Lifecycles { get; }
-        IMachinePolicyRepository MachinePolicies { get; }
-        IMachineRepository Machines { get; }
-        IMachineRoleRepository MachineRoles { get; }
-        IMigrationRepository Migrations { get; }
-        IOctopusServerNodeRepository OctopusServerNodes { get; }
-        IPerformanceConfigurationRepository PerformanceConfiguration { get; }
-        IProjectGroupRepository ProjectGroups { get; }
-        IProjectRepository Projects { get; }
-        IProjectTriggerRepository ProjectTriggers { get; }
-        IProxyRepository Proxies { get; }
-        IReleaseRepository Releases { get; }
-        IRetentionPolicyRepository RetentionPolicies { get; }
-        ISchedulerRepository Schedulers { get; }
-        IServerStatusRepository ServerStatus { get; }
-        ISubscriptionRepository Subscriptions { get; }
-        ITagSetRepository TagSets { get; }
+        IEventRepository Events { get; }        
         ITaskRepository Tasks { get; }
         ITeamsRepository Teams { get; }
-        ITenantRepository Tenants { get; }
-        ITenantVariablesRepository TenantVariables { get; }
-        IUserRepository Users { get; }
-        IUserRolesRepository UserRoles { get; }
-        IVariableSetRepository VariableSets { get; }
-        IWorkerPoolRepository WorkerPools { get; }
-        IWorkerRepository Workers { get; }
+        /// <summary>
+        /// Gets a document that identifies the Octopus server (from /api) and provides links to the resources available on the
+        /// server. Instead of hardcoding paths,
+        /// clients should use these link properties to traverse the resources on the server. This document is lazily loaded so
+        /// that it is only requested once for
+        /// the current <see cref="IOctopusAsyncClient" />.
+        /// </summary>
+        /// <exception cref="OctopusSecurityException">
+        /// HTTP 401 or 403: Thrown when the current user's API key was not valid, their
+        /// account is disabled, or they don't have permission to perform the specified action.
+        /// </exception>
+        /// <exception cref="OctopusServerException">
+        /// If any other error is successfully returned from the server (e.g., a 500
+        /// server error).
+        /// </exception>
+        /// <exception cref="OctopusValidationException">HTTP 400: If there was a problem with the request provided by the user.</exception>
+        /// <exception cref="OctopusResourceNotFoundException">HTTP 404: If the specified resource does not exist on the server.</exception>
+        RootResource RootDocument { get; }
```

Comment:
I would put this on `IAsyncSystemRepository`. Consider customers that have disabled the default space: they should ideally never see or use this particular repository (`IOctopusAsyncRepository `) and should instead use the system/space variants instead.